Bender.yml

```diff
@@ -149,13 +149,17 @@ sources:
       - hw/snitch_ssr/test/tb_simple_ssr.sv
       - hw/snitch_ssr/test/tb_simple_ssr_streamer.sv
 
+  # cluster peripherals
+  - target: snitch_cluster
+    files:
+      - target/snitch_cluster/generated/snitch_cluster_peripheral_reg_pkg.sv
+      - target/snitch_cluster/generated/snitch_cluster_peripheral_reg_top.sv
+      - hw/snitch_cluster/src/snitch_cluster_peripheral/snitch_cluster_peripheral.sv
+
```

Comment:
There's actually already a `snitch_cluster` target, see below. But I'm not sure we can use the same target for both. Consider e.g. Occamy, where we actually don't want to include the `snitch_cluster` target with the `snitch_cluster_wrapper` file, cause it generates its own cluster wrapper.

Is this the same also in the FlooNoC-based system?

Reply:
This is actually the only reason why we didn't add the `snitch_cluster` target directly to the `COMMON_BENDER_FLAGS` in the `common.mk` file. As this would be used by derived systems e.g. Occamy, but those don't necessarily include `snitch_cluster_wrapper.sv`.

Reply:
I see, in that case I will not specify a target for the peripheral registers. I thought it might make sense to "bundle" the auto-generated sources with the `snitch_cluster` target

Reply:
The problem then might be that derived systems first need to generate the regfile sources if they are not being tracked. Maybe it would make more sense to keep them in the `snitch_cluster_peripheral` directory and track them. What do you think?

Reply:
Perhaps it would make sense, maybe for derived systems, but let's see then when we have a need and use-case for this. Just we would have to have two distinct targets.

Reply:
Yes indeed, I don't know what the best approach is in this regard.

1) Probably the simplest is to just generate them in `PERIPH_DIR` and track them (but ideally the CI would then check that the tracked files are always up-to-date with the generated ones).

2) The best way would probably be to do like the DMA, where the files are generated and distributed with releases, but this is more complex to set up, and I guess it relies on versioning.

3) A compromise solution could be to not track them and let derived systems generate them. Anyways, they would ideally reuse the Make snippets provided in `common`. We could think of just moving the peripherals generation there.

Maybe we can just go with the first for now, to start merging something. But I'm also good with the third, as you prefer.

Reply:
I went with the first option for now, and the generated regfiles are tracked in the `PERIPH_DIR` folder. I also added a CI diff porcelain check to make sure the sources are always up to date.